src/layman/layer/db/__init__.py

```diff
@@ -135,7 +149,7 @@ def check_new_layername(workspace, layername, conn_cur=None):
         cur.execute(f"""SELECT n.nspname AS schemaname, c.relname, c.relkind
     FROM   pg_class c
     JOIN   pg_namespace n ON n.oid = c.relnamespace
-    WHERE  n.nspname IN ('{workspace}', '{settings.PG_POSTGIS_SCHEMA}') AND c.relname='{layername}'""")
+    WHERE  n.nspname IN ('{workspace}', '{settings.PG_POSTGIS_SCHEMA}') AND c.relname='{table_name}'""")
     except BaseException as exc:
         logger.error(f'check_new_layername ERROR')
         raise LaymanError(7) from exc
```

Comment:
LaymanError 7 is maybe not needed any more
LaymanError 9 is maybe not needed any more